### ISSUE TITLE
Segmentation fault after latest update

### ISSUE THREAD

**Issue by snomos** [divvun/gut#179]
```sh
gut clone -r corpus
zsh: segmentation fault  gut clone -r corpus
gut clone -u -r corpus -o giellalt
zsh: segmentation fault  gut clone -u -r corpus -o giellalt
gut pull -r .
zsh: segmentation fault  gut pull -r .
```

The previous version (from December 2022) does not crash.

**Comment by snomos** [comment]
After commit 1c2cf43f5346e8f85ba658de85018f4265f693fb, it still segfaults, but now it becomes clear that it happens at the very end of the processing:

```sh
gut pull -r . -o divvun
 2023-04-18T06:38:11.632Z DEBUG gut > Arguments: Args { command: Pull(PullArgs { organisation: Some("divvun"), regex: Some(Filter { regex: . }), stash: false, merge: false }) }
 2023-04-18T06:38:11.635Z INFO  gut::commands::pull > Processing repo divvun-manager-macos
[...] 
 2023-04-18T06:38:15.044Z INFO  gut::commands::pull > Processing repo satni-frontend
 2023-04-18T06:38:15.049Z INFO  gut::commands::pull > Processing repo woodwing-divvungc
 2023-04-18T06:38:15.100Z INFO  gut::commands::pull > Processing repo xkb-parser
[...] 
zsh: segmentation fault  gut pull -r . -o divvun
```

**Comment by snomos** [comment]
Something in 6f65edd3a2f514a2d95fa3851fb21d1b89e45449 is causing a segfault, but I really don't see what it could be.

**Comment by snomos** [comment]
I ran:

```sh
cargo install -f cargo-upgrades
cargo install -f cargo-edit
```

Then the following command:

```sh
cargo upgrades
gut: /Users/smo036/langtech/gut/divvun/gut/Cargo.toml
	base64 matches 0.12.3;	latest is 0.21.0
	clap matches 2.34.0;	latest is 4.2.2
	color-backtrace matches 0.4.2;	latest is 0.5.1
	colored matches 1.9.3;	latest is 2.0.0
	dialoguer matches 0.5.1;	latest is 0.10.4
	dirs matches 2.0.2;	latest is 5.0.0
	git2 matches 0.13.25;	latest is 0.17.1
	git2_credentials matches 0.6.1;	latest is 0.11.0
	graphql_client matches 0.9.0;	latest is 0.12.0
	prettytable-rs matches 0.8.0;	latest is 0.10.0
	reqwest matches 0.10.10;	latest is 0.11.16
	toml matches 0.5.11;	latest is 0.7.3
	uuid matches 0.8.2;	latest is 1.3.1
	proptest matches 0.9.6;	latest is 1.1.0
```

followed by this command:

```sh
cargo upgrade              
    Updating 'https://github.com/rust-lang/crates.io-index' index
    Checking gut's dependencies
warning: ignoring reqwest, unsupported entry: Use of `default_features` in `reqwest` is unsupported, please switch to `default-features`
name             old req compatible latest new req note        
====             ======= ========== ====== ======= ====        
toml             0.5.6   0.5.11     0.7.3  0.5.11  incompatible
structopt        0.3.9   0.3.26     0.3.26 0.3.26              
clap             2.0     2.34.0     4.2.2  2.34    incompatible
thiserror        1.0.11  1.0.40     1.0.40 1.0.40              
anyhow           1.0.26  1.0.70     1.0.70 1.0.70              
log              0.4.8   0.4.17     0.4.17 0.4.17              
color-backtrace  0.4.0   0.4.2      0.5.1  0.4.2   incompatible
prettytable-rs   0.8.0   0.8.0      0.10.0 0.8.0   incompatible
colored          1.9.3   1.9.3      2.0.0  1.9.3   incompatible
graphql_client   0.9.0   0.9.0      0.12.0 0.9.0   incompatible
dirs             2.0     2.0.2      5.0.0  2.0     incompatible
regex            1.3.4   1.7.3      1.7.3  1.7.3               
git2             0.13.6  0.13.25    0.17.1 0.13.25 incompatible
git2_credentials 0.6.0   0.6.1      0.11.0 0.6.1   incompatible
dialoguer        0.5.0   0.5.1      0.10.4 0.5.1   incompatible
base64           0.12.1  0.12.3     0.21.0 0.12.3  incompatible
uuid             0.8     0.8.2      1.3.1  0.8     incompatible
rayon            1.1     1.7.0      1.7.0  1.7                 
proptest         0.9.5   0.9.6      1.1.0  0.9.6   incompatible
tempfile         3.1.0   3.5.0      3.5.0  3.5.0               

   Upgrading recursive dependencies
note: Re-run with `--incompatible` to upgrade incompatible version requirements
note: Re-run with `--verbose` to show all dependencies
  unchanged: dryoc, pretty_env_logger, serde, serde_json, walkdir
```

**Comment by snomos** [comment]
That is, there seems to be a lot of outdated deps, and many of them incompatible with the present codebase. It will take some work to fix all this. We don't know whether fixing them will also fix the segfault, but it is not unlikely - at least the git deps include some C++ libs in the background that will force `unsafe`, and thus be at risk for triggering segfaults.

**Comment by flammie** [comment]
yeah I tried some updates like that when working on these things few weeks ago but at that time but couldn't get cargo to find dependencies that would even compile so I gave up... I'll see if there¨s a way around now

**Comment by snomos** [comment]
You could have a look at @bbqsrc ’s PR. It builds but is otherwise completely untested.

**Comment by snomos** [comment]
Fixed by e5bfd89910708f12aa257dda25b9bfb505a8733a. All deps are updated. Option to set https as default is lost, and must be reimplemented. @flammie ?
